Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/submission-box/SubmissionBox.tsx

```diff
@@ -195,20 +201,36 @@ const SubmissionBox = () => {
         executionTypeListClass,
     );
 
+    const submissionCode = useMemo(
+        () => {
+            const { id: problemId } = currentProblem || {};
+            if (isNil(problemId)) {
+                return null;
+            }
+            return problemSubmissionCode[problemId];
+        },
+        [ currentProblem, problemSubmissionCode ],
+    );
+
     const renderSubmissionInput = useCallback(() => {
         if (isNil(selectedSubmissionType)) {
             return <p>No submission type selected.</p>;
         }
 
         const { allowBinaryFilesUpload, allowedFileExtensions } = selectedSubmissionType;
+        const { id: problemId } = currentProblem || {};
+        if (isNil(problemId)) {
+            return null;
+        }
 
-        if (allowBinaryFilesUpload) {
+        if (allowBinaryFilesUpload && !isNil(currentProblem)) {
             return (
                 <>
-                    <FormControl
-                      type={FormControlType.file}
-                      name="file"
-                      onChange={(file) => handleCodeChanged(file as Blob)}
+                    <FileUploader
+                      file={isNil(submissionCode) || submissionCode instanceof String
```

Comment:
Is this check needed 'submissionCode instanceof String' 
If allowBinaryFilesUpload is true and submissionCode is not null you can directly return submissionCode as File
What are your thoughts on this and what is the purpose of the check?

Reply:
Submissioncode for a problem can be either a string or a file. If it is a string and we click on C#project/solution strategy, then without the check we are sending a string to the file uploader component, which expects to receive a file and the app crashes.